### PR TITLE
Auto-wiki setup for soup

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Sets up an auto-wiki run for Soup.

It's all in the in game cookbook, but keeping the wiki up to date is good. 

Still figuring out how auto-wiki works with the help of Smithers, so it might need to be updated in a follow-up PR. 

```
{{Autowiki/SoupRecipeTemplate
|name=Tomato Soup
|taste=tomato
|foodtypes=Vegetables, Fruits
|description=Drinking this feels like being a vampire! A tomato vampire...
|icon=soup/Tomato Soup
|requirements=50 units Water, 2 tomatos, at temperature 450K
|results=30 units Tomato Soup, 20 units Tomato Juice
}}
```

Templates: 
https://tgstation13.org/wiki/Template:Autowiki/SoupRecipeTemplate
https://tgstation13.org/wiki/Template:Autowiki/SoupRecipeTableTemplate

### CODE REVIEW

**Comment by Mothblocks** [comment]
I unfortunately never got around to making Autowiki intuitive to use, though I'm not sure how much Smithers knows. When you get this tested and complete, can you link the article to the template page?

**Comment by SpaceSmithers** [comment]
I don't know squat about autowiki code-wise, but given its outputs I'm happy to help create a good looking and functional frontend for the wiki.

**Comment by Mothblocks** [comment]
did you figure out file uploading?

**Comment by Mothblocks** [comment]
The icons from vending machines are uploaded as part of autowiki. It looks like you want images in your templates when I checked?

**Comment by Mothblocks** [comment]
but its not being uploaded by the autowiki script? it definitely should...

i am out on a business trip but if you can't figure it out by then i'll figure out why that's not working. or are you not actually using the script and you're just generating templates and uploading them yourself?

**Comment by Mothblocks** [comment]
Fair enough. I'll run this tonight

File: code/modules/autowiki/pages/soup.dm [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	qdel(soup_pot)
+
+	var/container_for_images = /obj/item/reagent_containers/cup/bowl
+
+	for(var/soup_recipe_type in subtypesof(/datum/chemical_reaction/food/soup))
+		var/datum/chemical_reaction/food/soup/soup_recipe = new soup_recipe_type()
+		var/result_soup_type = soup_recipe.results[1]
+		var/datum/reagent/result_soup = new result_soup_type()
+		var/datum/glass_style/has_foodtype/soup_style = GLOB.glass_style_singletons[container_for_images][result_soup_type]
+		var/filename = "soup/[SANITIZE_FILENAME(escape_value(format_text(result_soup.name)))]"
```
Comment (Mothblocks) [inline code/modules/autowiki/pages/soup.dm:24]:
```suggestion
		var/filename = "soup_[SANITIZE_FILENAME(escape_value(format_text(result_soup.name)))]"
```

Files can't namespace

**Comment by Mothblocks** [comment]
It works after that one edit but I want Melbert to write some code to resize them to be bigger

**Pull request merged**